src/PolyType/GenerateShapeAttribute.cs

```diff
@@ -3,19 +3,54 @@
 namespace PolyType;
 
 /// <summary>
-/// Instructs the PolyType source generator to include the annotated type
+/// Instructs the PolyType source generator to a given type
 /// in the <see cref="ITypeShapeProvider"/> that it generates.
 /// </summary>
 /// <remarks>
+/// <para>
 /// For projects targeting .NET 8 or later, this additionally augments the type
 /// with an implementation of IShapeable for the type.
-///
+/// </para>
+/// <para>
 /// Projects targeting older versions of .NET need to access the generated
 /// <see cref="ITypeShapeProvider"/> instance through the static property
-/// added to classes annotated with the <see cref="GenerateShapeAttribute{T}"/>.
+/// added to classes annotated with the <see cref="GenerateShapeAttribute{T}"/>
+/// or <see cref="GenerateShapeAttribute(Type)"/>.
+/// </para>
 /// </remarks>
-[AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface, AllowMultiple = false, Inherited = false)]
-public sealed class GenerateShapeAttribute : Attribute;
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface, AllowMultiple = true, Inherited = false)]
```

Comment:
Why `AllowMultiple = true`?

Reply:
Can we somehow keep those responsibilities distinct? E.g. by introducing a new attribute with a separate name and/or renaming some of the existing ones?

Reply:
Because this has to be allowed:

```cs
[GenerateShape(typeof(Foo))]
[GenerateShape(typeof(Bar))]
partial class Witness;
```

Before, with no constructor parameters, there was never a need for the attribute to appear multiple times.

The only way to avoid this is to give the witness-targeted attribute a new name. e.g. `GenerateWitnessShapeForAttribute<T>` and `GenerateWitnessShapeForAttribute(Type)`. I elected to not go for that. But if you'd prefer that, we can do that.

Reply:
Your follow-up comments appeared only after I submitted my first reply. 
What would you like to rename the generic attribute and this new non-generic witness attribute to?

Reply:
Here's one idea:
```
[GenerateShape] // Generates for the current type
public partial class Foo;

[GenerateShapeOf(typeof(Foo))]
[GenerateShapeOf<Foo>]
public partial class Witness;
```
Naming needs a bit of work, but structurally I think this is where we need to head towards.

Reply:
Renamed to `GenerateShapeFor`.